Sources/AccessibilitySnapshot/Core/Swift/Classes/HitTargetSnapshotView.swift

```diff
@@ -0,0 +1,141 @@
+    /// - parameter colors: An array of colors to use for the highlighted regions. These colors will be used in order,
+    /// repeating through the array as necessary and avoiding adjacent regions using the same color when possible.
+    public static func generateSnapshotImage(
+        for view: UIView,
+        useMonochromeSnapshot: Bool,
+        viewRenderingMode: AccessibilitySnapshotView.ViewRenderingMode,
+        colors: [UIColor] = AccessibilitySnapshotView.defaultMarkerColors
+    ) throws -> UIImage {
+        let colors = colors.map { $0.withAlphaComponent(0.2) }
+
+        let bounds = view.bounds
+        let renderer = UIGraphicsImageRenderer(bounds: bounds)
+
+        let viewImage = try view.renderToImage(
+            monochrome: useMonochromeSnapshot,
+            viewRenderingMode: viewRenderingMode
+        )
+
+        return renderer.image { context in
+            viewImage.draw(in: bounds)
+
+            var viewToColorMap: [UIView: UIColor] = [:]
+            let pixelWidth: CGFloat = 1 / UIScreen.main.scale
+
+            func drawScanLine(
+                for hitView: UIView?,
+                startingAtX: CGFloat,
+                endingAtX: CGFloat,
+                y: CGFloat,
+                lineHeight: CGFloat
+            ) {
+                // Only draw hit areas for views other than the base view we're testing.
+                guard hitView !== view else {
+                    return
+                }
+
+                let color: UIColor
+                if let hitView = hitView, let existingColor = viewToColorMap[hitView] {
+                    color = existingColor
+                } else if let hitView = hitView {
+                    // As a future enhancement, this could be smarter about checking above/left colors to make sure they
+                    // aren't the same.
+                    color = colors[viewToColorMap.count % colors.count]
+                    viewToColorMap[hitView] = color
+                } else {
+                    color = .lightGray
+                }
+
+                context.cgContext.setFillColor(color.cgColor)
+                context.cgContext.beginPath()
+                context.cgContext.addRect(
+                    CGRect(
+                        x: startingAtX,
+                        y: y,
+                        width: (endingAtX - startingAtX),
+                        height: lineHeight
+                    )
+                )
+                context.cgContext.drawPath(using: .fill)
+            }
+
+            let touchOffset = pixelWidth / 2
+
+            // Step through every pixel along the Y axis.
+            for y in stride(from: bounds.minY, to: bounds.maxY, by: pixelWidth) {
+                var lastHit: (CGFloat, UIView?)? = nil
+
+                // Step through every pixel along the X axis.
+                for x in stride(from: bounds.minX, to: bounds.maxX, by: pixelWidth) {
```

Comment:
I imagine this nested loop is a factor in the slow performance you commented on above. If we knew beforehand that all of the hit targets in question were rectangular, we could probably optimize this to just look for the overall width/height of the target, but this scanning seems necessary for anything non-rectangular.

Reply:
...sure enough, this is addressed in #177

Reply:
Yep, exactly. Guaranteed rectangular hit targets would make this way easier, but given how flexible the hit test APIs are, this is the only way I can think of to ensure we always get it correct.